extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/KinesisRecordSupplier.java

```diff
@@ -663,32 +661,33 @@ public String getEarliestSequenceNumber(StreamPartition<String> partition)
     return getSequenceNumber(partition, ShardIteratorType.TRIM_HORIZON);
   }
 
+  /**
+   * Use the API listShards which is the recommended way instead of describeStream
+   * listShards can return 1000 shards per call and has a limit of 100TPS
+   * This makes the method resilient to LimitExceeded exceptions (compared to 100 shards, 10 TPS of describeStream)
+   *
+   * @param stream name of stream
+   *
+   * @return Set of Shard ids
+   */
   @Override
   public Set<String> getPartitionIds(String stream)
   {
     return wrapExceptions(
         () -> {
           final Set<String> retVal = new HashSet<>();
-          DescribeStreamRequest request = new DescribeStreamRequest();
-          request.setStreamName(stream);
-
-          while (request != null) {
-            final DescribeStreamResult result = kinesis.describeStream(request);
-            final StreamDescription streamDescription = result.getStreamDescription();
-            final List<Shard> shards = streamDescription.getShards();
-
+          ListShardsRequest request = new ListShardsRequest().withStreamName(stream);
+          while (true) {
+            ListShardsResult result = kinesis.listShards(request);
+            List<Shard> shards = kinesis.listShards(request).getShards();
```

Comment:
1. remove the double call
2. checking for `result.getNextToken() == null` instead of empty list seems better
3. while creating the new request in loop, set the stream name as well
4. <nit> : can try to do a bulk insert inside the `retVal`

Reply:
Points 1, 2 and 4 have been addressed. 
Token and streamName cannot be provided simultaneously to a ListShardsRequest